### PR TITLE
Updated Javascript.json/Modified the body

### PR DESCRIPTION
**Pull request by Priyanshu-bit**
Updated CreateMap, CreateScene,CreateWebMap, CreateWebScene. Removed assign variables after class Initialization.

### CODE REVIEW

**Comment by hhkaos** [comment]
Thanks @Priyanshu-bit, I just looked at it and the changes are good. But as you described, this PR includes one commit to apply only one [body convention](https://github.com/Esri/arcgis-js-vscode-snippets/blob/master/CONTRIBUTING.md#body) -> [After a class initialization, don't assign it to a variable](https://github.com/Esri/arcgis-js-vscode-snippets/blob/master/CONTRIBUTING.md#5-after-a-class-initialization-dont-assign-it-to-a-variable).

I'm missing fixes related to other conventions, for example: 

```
"CalloutLarge": {
    "body": [
      "verticalOffset: {",
      "\tscreenLength: 40,",
      "\tmaxWorldLength: 500000,",
      "\tminWorldLength: 0",
      "},",
      "callout: {",
      "\ttype: \"line\",",
      "\tsize: ${1:1.5},",
      "\tcolor: ${2:[255, 255, 255, 0.9]},",
      "\tborder: {",
      "\t\tcolor: ${3:[0, 0, 0, 0.5]}",
      "\t}",
      "}"
    ],
    "description": "Generate 3D callouts for world scale",
    "prefix": "calloutLarge"
  },
```

In this case:
* [prefix conventions](https://github.com/Esri/arcgis-js-vscode-snippets/blob/master/CONTRIBUTING.md#prefix): It should include the sufix "Props" `calloutLargeProps`
* [name conventions](https://github.com/Esri/arcgis-js-vscode-snippets/blob/master/CONTRIBUTING.md#name). I this case, it is using "Pascal case". 
	* I would probably suggest adding one more bullet to the conventions saying: "Names should be descriptive and not overlap with the `prefix`. Description will extend this short description.". Otherwise we end-up with the Intellisense doing things like this: Screenshot 2023-09-28 at 07 39 53
	* I would probably use as name "Symbol 3D callout" or "Symbol 3D world-scale callout"

And body is not in alphabetical order. 

```js
verticalOffset: {
	screenLength: 40,
	maxWorldLength: 500000,
	minWorldLength: 0
},

callout: {
	type: "line",
	size: 1.5,
	color: [255, 255, 255, 0.9],
	border: {
		color: [0, 0, 0, 0.5]
	}
}
```

Use alphabetical order
```js
callout: {
	type: "line",
	size: 1.5,
	color: [255, 255, 255, 0.9],
	border: {
		color: [0, 0, 0, 0.5]
	}
}
verticalOffset: {
	screenLength: 40,
	maxWorldLength: 500000,
	minWorldLength: 0
},
```

But I noticed when didn't specify if nested objects should also be in alphabetical order. I guess we should comment in https://github.com/Esri/arcgis-js-vscode-snippets/issues/19

I hope this helps.

I'll wait for upcoming commits before merging this. Thanks again for your help!

**Comment by hhkaos** [comment]
I forgot to mention. I have also made some changes to the contributing guidelines to help you do better commit&pr messages.

Please check: https://github.com/Esri/arcgis-js-vscode-snippets/pull/38/files

**Comment by hhkaos** [comment]
One additional comment, I think the "map" snippet, should be refactored to use autocast, I mean:

```
new MapView({
  container: "viewDiv",
  map: {
    basemap: "streets"
  },
  zoom:  4,
  center: [15,65]
});
```
Otherwise, we will be forcing the developer to assign the Map class instance to a variable manually to be able to use it. Thoughts?

**Comment by Priyanshu-bit** [comment]
Thank you for your patience @hhkaos .I'm working in it.

**Review by hhkaos — CHANGES_REQUESTED** [review]
As I mentioned in the comments, these changes look pretty good, but `CreateMap` needs to be refactored in order to work.

**Comment by Priyanshu-bit** [comment]
> As I mentioned in the comments, these changes look pretty good, but `CreateMap` needs to be refactored in order to work.
Sure, will be done by the end of the day

File: snippets/javascript.json [hunk 1]
```diff
@@ -548,7 +552,7 @@
     "description": "Generate a UniqueValueRenderer",
     "prefix": "uniqueValueRenderer"
   },
-  "WaterSymbol3DLayer": {
+  " Generate water symbol 3D layer": {
     "prefix": "waterSymbol3dLayer",
```
Comment (hhkaos) [inline snippets/javascript.json:556]: waterSymbol3dLayerProps

File: snippets/javascript.json [hunk 2]
```diff
@@ -564,7 +568,7 @@
     ],
     "description": "Create PolygonSymbol3D with a WaterSymbol3DLayer"
   },
-  "Generate a DotDensity Renderer": {
+  "Generate a DotDensity renderer": {
     "prefix": "dotDensity",
```
Comment (hhkaos) [inline snippets/javascript.json:572]: dotDensityRendererProps

**Review by hhkaos — CHANGES_REQUESTED** [review]
Great work @Priyanshu-bit !

I just noticed you missed a some conventions like the "Props" suffix, or add the AMD & ES paths in the description when the script is initializing a class. 

Also check the `names`, same have spaces at the start/end, e.g:

* "Add a layer from a portal item " -> "Add a layer from a portal item"
*   " Generate water symbol 3D layer" ->   "Generate water symbol 3D layer"
* ...

Thanks again, this one is getting close to the goal! :)

File: snippets/javascript.json [hunk 3]
```diff
@@ -115,7 +116,7 @@
     "description": "Create map and scene view",
     "prefix": "scene"
   },
-  "CreateWebMap": {
+  "Create web map": {
     "body": [
       "new WebMap({",
```
[the remark this answers is not in the document]
Comment (Priyanshu-bit) [reply snippets/javascript.json:121]: Done!

File: snippets/javascript.json [hunk 4]
```diff
@@ -487,7 +491,7 @@
     "description": "Create a SimpleLineSymbol",
     "prefix": "simpleLineSymbol"
```
[the remark this answers is not in the document]
Comment (Priyanshu-bit) [reply snippets/javascript.json:492]: simpleLineSymbolProps ?

File: snippets/javascript.json [hunk 5]
```diff
@@ -102,7 +103,7 @@
     "prefix": "map"
   },
   
-  "CreateScene": {
+  "Create scene": {
     "body": [
       "new Map({",
       "\tbasemap: \"${1:streets}\"",
```
[the remark this answers is not in the document]
Comment (Priyanshu-bit) [reply snippets/javascript.json]: autocasting done, rest cannot understand properly

**Comment by Priyanshu-bit** [comment]
I want to express my sincere thanks  to @hhkaos  for the invaluable support and guidance you've provided throughout my journey. I'm absolutely thrilled about my recent contribution and eagerly look forward to further collaborations.

Your assistance has been instrumental, and I appreciate it immensely. Here's to more exciting contributions in the future!

**Comment by hhkaos** [comment]
I've also learned a lot during this process, so thank you very much @Priyanshu-bit. And thank you for all of your work and effort in this contribution; it is undeniably an essential contribution to the new path of this extension. I look forward to continuing to collaborate with you.

Thanks! 🙏

**Pull request merged**